File: app/Models/Incident.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\User;
use App\Models\Vehicle;
use App\Models\Park;

class Incident extends Model
{
    use HasFactory;

    // La clave primaria es id_incidencia, que no es autoincremental
    protected $primaryKey = 'id_incidencia';
    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'id_incidencia',
        'id_empleado',   // Creador de la incidencia
        'tipo',          // vehiculo, personal o instalacion
        'estado',
        'matricula',     // Si tipo es "vehiculo"
        'id_empleado2',  // Si tipo es "personal"
        'resulta_por',   // Empleado que resuelve la incidencia
        'leido',         // Bool que indica si ha sido leído por los jefes
        'id_parque',     // Parque donde se genera la incidencia
        'fecha',
        'descripcion'
    ];

    // Relación con el usuario creador de la incidencia.
    public function creator()
    {
        return $this->belongsTo(User::class, 'id_empleado');
    }

    // Si la incidencia es de tipo "vehiculo", se relaciona con el vehículo (usando la matrícula)
    public function vehicle()
    {
        return $this->belongsTo(Vehicle::class, 'matricula', 'matricula');
    }

    // Si la incidencia es de tipo "personal", se relaciona con el empleado referenciado.
    public function employee2()
    {
        return $this->belongsTo(User::class, 'id_empleado2');
    }

    // Empleado que resuelve la incidencia
    public function resolver()
    {
        return $this->belongsTo(User::class, 'resulta_por');
    }

    // Relación con el parque donde se origina la incidencia.
    public function park()
    {
        return $this->belongsTo(Park::class, 'id_parque');
    }
}

File: android/app/src/main/java/es/bomberosgranada/app/data/models/Incident.kt
package es.bomberosgranada.app.data.models

data class Incident(
    val id_incidencia: Int,
    val fecha: String,
    val id_empleado: Int,
    val tipo: String,
    val estado: String,
    val descripcion: String,
    val nivel: String,
    val id_parque: Int,
    val leido: Boolean,
    val resolviendo: String? = null,
    val matricula: String? = null,
    val id_empleado2: Int? = null,
    val equipo: String? = null,
    val id_vestuario: Int? = null,
    val resulta_por: Int? = null,
    val resolucion: String? = null,
    val created_at: String,
    val updated_at: String? = null,
    // Relaciones
    val creator: User? = null,
    val employee2: User? = null,
    val vehicle: Vehicle? = null,
    val park: Park? = null,
    val resolver: User? = null,
    val clothing_item: ClothingItem? = null,
    val equipment: PersonalEquipment? = null
)

data class CreateIncidentRequest(
    val fecha: String,
    val id_empleado: Int,
    val tipo: String,
    val estado: String,
    val descripcion: String,
    val nivel: String,
    val id_parque: Int,
    val leido: Boolean = false,
    val resolviendo: String? = null,
    val matricula: String? = null,
    val id_empleado2: Int? = null,
    val equipo: String? = null,
    val id_vestuario: Int? = null
)

data class UpdateIncidentRequest(
    val fecha: String? = null,
    val id_empleado: Int? = null,
    val tipo: String? = null,
    val estado: String? = null,
    val descripcion: String? = null,
    val nivel: String? = null,
    val id_parque: Int? = null,
    val leido: Boolean? = null,
    val resolviendo: String? = null,
    val matricula: String? = null,
    val id_empleado2: Int? = null,
    val equipo: String? = null,
    val id_vestuario: Int? = null
)

data class ResolveIncidentRequest(
    val resulta_por: Int,
    val resolucion: String
)

data class CountResponseIncident(
    val pending: Int
)

data class Vehicle(
    val id: Int,
    val matricula: String,
    val tipo: String? = null
)

